rust/core-lib/src/file.rs

```diff
@@ -105,9 +105,9 @@ impl FileManager {
         false
     }
 
-    pub fn open(&mut self, path: &Path, id: BufferId) -> Result<Rope, FileError> {
+    pub fn open(&mut self, path: &Path, id: BufferId) -> Result<Option<Rope>, FileError> {
         if !path.exists() {
-            let _ = File::create(path).map_err(|e| FileError::Io(e, path.to_owned()))?;
+            return Ok(None);
```

Comment:
We could also just return an empty rope here? I don't have a strong opinion, returning an empty rope is a smaller change. It's possible that at some point we will want to have some special behaviour if the file doesn't exist, but since we don't do that now I think we can do whatever is easiest.

Reply:
Alrighty, that can work too, but felt a bit dirty to me. I'm fine with this though.

Reply:
Pushed new version.